src/app/services/movies/movies.service.ts

```diff
@@ -49,4 +41,15 @@ export class MoviesService {
   findMovie(title: string) {
     this.movies = this.movies.filter(e => e.title === title);
   }
+
+  editMovie(movie: Movie) {
+    this.http.put("http://localhost:4000/movies", movie).subscribe(
+      e => console.log(e)
+    )
+    const movieIndex = this.movies.findIndex((m) => m.id === movie.id)
+    if (movieIndex < 0) {
```

Comment:
something odd here, you're editing a movie which means that the movie is in this.movies so that's why you are able to edit it, but you're checking if the movie index exist